roundabout-core/src/main/java/at/fhv/itm3/s2/roundabout/entity/StreetSection.java

```diff
@@ -66,12 +66,31 @@ public boolean carCouldEnterNextSection() {
         throw new NotImplementedException();
     }
 
+
+    /**
+     * moveFirstCarToNextSection removes the first car from the queue and puts it into the
+     * queue of the next streetSection of the route of the car, if there is one.
+     * If the current streetSection was the last one of the route the car disappears in a sink.
+     */
     public void moveFirstCarToNextSection() {
-        throw new NotImplementedException();
+        Car firstCar = (Car)carQueue.poll();
+        if (firstCar!=null){
+            StreetSection currentSection = firstCar.getCurrentSection();
+            int currentSectionIndex = firstCar.getRoute().indexOf(currentSection);
+            if (currentSectionIndex < (firstCar.getRoute().size()-1)){
```

Comment:
you could use getDestination() from car, to get the destination section
just define the method in ICar interface

Reply:
What is the differnce between getDestination and getNextStreetSection?

Reply:
getDestination always returns the last section in the cars route